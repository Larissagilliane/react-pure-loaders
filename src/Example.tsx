import * as React from 'react';

export const Example = (props: any) => {
  return <span {...props}>Hello World!</span>;
};