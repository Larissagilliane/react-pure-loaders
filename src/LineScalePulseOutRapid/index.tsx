import { css } from 'glamor';
import * as React from 'react';
import style from './style';

export interface IGeneralPropTypes {
  loading: boolean;
}

export class LineScalePulseOutRapid extends React.PureComponent<IGeneralPropTypes, {}> {

  public render() {
    const { loading } = this.props;

    if (loading) {
      return (
        <div {...css(style)}>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
        </div>
      );
    }

    return null;
  }
}
