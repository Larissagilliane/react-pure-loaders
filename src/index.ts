/**
 * Dots
 */
export { BallBeat } from './BallBeat';
export { BallClipRotate } from './BallClipRotate';
export { BallClipRotateMultiple } from './BallClipRotateMultiple';
export { BallClipRotatePulse } from './BallClipRotatePulse';
export { BallGridBeat } from './BallGridBeat';
export { BallGridPulse } from './BallGridPulse';
export { BallPulse } from './BallPulse';
export { BallPulseRise } from './BallPulseRise';
export { BallPulseRound } from './BallPulseRound';
export { BallPulseSync } from './BallPulseSync';
export { BallRotate } from './BallRotate';
export { BallScale } from './BallScale';
export { BallScaleMultiple } from './BallScaleMultiple';
export { BallScaleRandom } from './BallScaleRandom';
export { BallScaleRipple } from './BallScaleRipple';
export { BallScaleRippleMultiple } from './BallScaleRippleMultiple';
export { BallSpinFadeLoader } from './BallSpinFadeLoader';
export { BallSpinLoader } from './BallSpinLoader';
export { BallTrianglePath } from './BallTrianglePath';
export { BallZigZag } from './BallZigZag';
export { BallZigZagDeflect } from './BallZigZagDeflect';

/**
 * Lines
 */
export { LineScale } from './LineScale';
export { LineScaleParty } from './LineScaleParty';
export { LineScalePulseOut } from './LineScalePulseOut';
export { LineScalePulseOutRapid } from './LineScalePulseOutRapid';
export { LineScaleRandom } from './LineScaleRandom';
export { LineSpinFadeLoader } from './LineSpinFadeLoader';

/**
 * Misc
 */
export { CubeTransition } from './CubeTransition';
export { Pacman } from './Pacman';
export { SemiCircleSpin } from './SemiCircleSpin';
export { SquareSpin } from './SquareSpin';
export { TriangleSkewSpin } from './TriangleSkewSpin';
