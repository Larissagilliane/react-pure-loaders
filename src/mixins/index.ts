import { css } from 'glamor';
import {
  BALL_SIZE,
  LINE_HEIGHT,
  LINE_WIDTH,
  MARGIN,
  PRIMARY_COLOR,
} from '../variables';

interface IGrid {
  [id: string]: object;
}

export function globalBg() {
  return {
    backgroundColor: PRIMARY_COLOR,
  };
}

export function globalAnimation() {
  return {
    animationFillMode: 'both',
  };
}

export function balls() {
  return css(
    {
      borderRadius: '100%',
      height: BALL_SIZE,
      margin: MARGIN,
      width: BALL_SIZE,
    },
    globalBg(),
  );
}

export function ballGrid() {
  let i = 1;
  const grid: IGrid = {};
  for (; i < 10; i += 1) {
    const delay = ((parseInt((Math.random() * 100).toString(), 10) / 100) - 0.2);
    const duration = ((parseInt((Math.random() * 100).toString(), 10) / 100) + 0.6);

    grid[` > div:nth-child(${i})`] = {
      animationDelay: `${delay}s`,
      animationDuration: `${duration}s`,
    };
  }

  return grid;
}

export function lines() {
  return css(
    {
      borderRadius: '2px',
      height: LINE_HEIGHT,
      margin: MARGIN,
      width: LINE_WIDTH,
    },
    globalBg(),
  );
}
