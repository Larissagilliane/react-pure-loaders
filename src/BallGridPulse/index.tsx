import { css } from 'glamor';
import * as React from 'react';
import { IGeneralPropTypes } from '../mixins';
import style from './style';

export class BallGridPulse extends React.PureComponent<IGeneralPropTypes, {}> {

  public render() {
    const { color, loading } = this.props;

    if (loading) {
      return (
        <div {...css(style(color))}>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
        </div>
      );
    }

    return null;
  }
}
