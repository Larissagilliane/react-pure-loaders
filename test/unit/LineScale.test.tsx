import { mount, render, shallow } from 'enzyme';
import { StyleSheet } from 'glamor/lib/sheet';
import * as React from 'react';
import { LineScale } from '../../src';

const stylesheet = new StyleSheet();

describe('<LineScale>', () => {
  beforeEach(() => {
    stylesheet.flush();
  });

  test('<LineScale> Shallow', () => {
    const ui = (<LineScale loading={true}/>);

    expect(shallow(ui)).toMatchSnapshot(`enzyme.shallow`);
  });
  test('<LineScale> Mount', () => {
    const ui = (<LineScale loading={true}/>);

    expect(mount(ui)).toMatchSnapshot(`enzyme.mount`);
  });
  test('<LineScale> Render', () => {
    const ui = (<LineScale loading={true}/>);

    expect(render(ui)).toMatchSnapshot(`enzyme.render`);
  });
});
