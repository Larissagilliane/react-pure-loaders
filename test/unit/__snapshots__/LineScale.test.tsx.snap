// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineScale> <LineScale> Mount: enzyme.mount 1`] = `
.css-0> div:nth-child(1),
[data-css-0]> div:nth-child(1) {
  animation: animation_19oh8vm 1s -0.40s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.40s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(2),
[data-css-0]> div:nth-child(2) {
  animation: animation_19oh8vm 1s -0.30s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.30s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(3),
[data-css-0]> div:nth-child(3) {
  animation: animation_19oh8vm 1s -0.20s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.20s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(4),
[data-css-0]> div:nth-child(4) {
  animation: animation_19oh8vm 1s -0.10s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.10s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(5),
[data-css-0]> div:nth-child(5) {
  animation: animation_19oh8vm 1s 0.00s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.00s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(6),
[data-css-0]> div:nth-child(6) {
  animation: animation_19oh8vm 1s 0.10s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.10s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(7),
[data-css-0]> div:nth-child(7) {
  animation: animation_19oh8vm 1s 0.20s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.20s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(8),
[data-css-0]> div:nth-child(8) {
  animation: animation_19oh8vm 1s 0.30s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.30s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(9),
[data-css-0]> div:nth-child(9) {
  animation: animation_19oh8vm 1s 0.40s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.40s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0 > div,
[data-css-0] > div {
  border-radius: 2px;
  height: 35px;
  margin: 2px;
  width: 4px;
  background-color: #ffffff;
  animation-fill-mode: both;
  display: inline-block;
  -webkit-animation-fill-mode: both;
}

<LineScale
  loading={true}
>
  <div
    data-css-0=""
  >
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</LineScale>
`;

exports[`<LineScale> <LineScale> Render: enzyme.render 1`] = `
.css-0> div:nth-child(1),
[data-css-0]> div:nth-child(1) {
  animation: animation_19oh8vm 1s -0.40s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.40s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(2),
[data-css-0]> div:nth-child(2) {
  animation: animation_19oh8vm 1s -0.30s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.30s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(3),
[data-css-0]> div:nth-child(3) {
  animation: animation_19oh8vm 1s -0.20s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.20s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(4),
[data-css-0]> div:nth-child(4) {
  animation: animation_19oh8vm 1s -0.10s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.10s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(5),
[data-css-0]> div:nth-child(5) {
  animation: animation_19oh8vm 1s 0.00s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.00s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(6),
[data-css-0]> div:nth-child(6) {
  animation: animation_19oh8vm 1s 0.10s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.10s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(7),
[data-css-0]> div:nth-child(7) {
  animation: animation_19oh8vm 1s 0.20s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.20s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(8),
[data-css-0]> div:nth-child(8) {
  animation: animation_19oh8vm 1s 0.30s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.30s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(9),
[data-css-0]> div:nth-child(9) {
  animation: animation_19oh8vm 1s 0.40s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.40s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0 > div,
[data-css-0] > div {
  border-radius: 2px;
  height: 35px;
  margin: 2px;
  width: 4px;
  background-color: #ffffff;
  animation-fill-mode: both;
  display: inline-block;
  -webkit-animation-fill-mode: both;
}

<div
  data-css-0=""
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`<LineScale> <LineScale> Shallow: enzyme.shallow 1`] = `
.css-0> div:nth-child(1),
[data-css-0]> div:nth-child(1) {
  animation: animation_19oh8vm 1s -0.40s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.40s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(2),
[data-css-0]> div:nth-child(2) {
  animation: animation_19oh8vm 1s -0.30s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.30s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(3),
[data-css-0]> div:nth-child(3) {
  animation: animation_19oh8vm 1s -0.20s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.20s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(4),
[data-css-0]> div:nth-child(4) {
  animation: animation_19oh8vm 1s -0.10s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s -0.10s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(5),
[data-css-0]> div:nth-child(5) {
  animation: animation_19oh8vm 1s 0.00s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.00s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(6),
[data-css-0]> div:nth-child(6) {
  animation: animation_19oh8vm 1s 0.10s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.10s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(7),
[data-css-0]> div:nth-child(7) {
  animation: animation_19oh8vm 1s 0.20s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.20s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(8),
[data-css-0]> div:nth-child(8) {
  animation: animation_19oh8vm 1s 0.30s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.30s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0> div:nth-child(9),
[data-css-0]> div:nth-child(9) {
  animation: animation_19oh8vm 1s 0.40s infinite cubic-bezier(.2,.68,.18,1.08);
  -webkit-animation: animation_19oh8vm 1s 0.40s infinite cubic-bezier(.2,.68,.18,1.08);
}

.css-0 > div,
[data-css-0] > div {
  border-radius: 2px;
  height: 35px;
  margin: 2px;
  width: 4px;
  background-color: #ffffff;
  animation-fill-mode: both;
  display: inline-block;
  -webkit-animation-fill-mode: both;
}

<div
  data-css-0=""
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
